### ISSUE TITLE
GELF Message is invalid received on stacktrace

### ISSUE THREAD

**Issue by FireballDWF** [mp911de/logstash-gelf#75]
Can't figure out why receiving "GELF Message is invalid:" as seen in the attached.

[gelf.txt](https://github.com/mp911de/logstash-gelf/files/190254/gelf.txt)

logging.properties file available at https://github.com/FireballDWF/docker-alpine-payara-logstash/blob/master/logging.properties and Dockerfile available at https://github.com/FireballDWF/docker-alpine-payara-logstash/blob/master/Dockerfile

**Comment by mp911de** [comment]
Hi @FireballDWF 
the message does not contain `short_message`/`full_message` fields. This happens when just the exception is logged without any additional details:

```
catch(Exception e) {
    logger.error(e);
}
```

The `GELF Message is invalid` should disappear with:


```
catch(Exception e) {
    logger.error(e.getMessage(), e);
}
```

Please let me know whether this works for you.

**Comment by FireballDWF** [comment]
@mp911de Thanks for creating a veru useful package and responding to the issue.  

If the change you propose only had to be done in my code, I would go ahead and do it.  But the exception is being logged by code from the EclipseLink project.  And anyone else using third party libraries which only logs the exception will experience the same impact (logs which won't make it to logstash without changing source).  

So would you be willing to change your library support this use case?  Even if it just substitutes an empty string?

**Comment by mp911de** [comment]
The empty message rule comes from the GELF spec. The only viable way to deal with it would be using `Throwable.getMessage()` or `Throwable.toString()` in case the original message is empty. Not sure whether I'm happy with this, but I'll consider it.

**Comment by mp911de** [comment]
I pushed a change on the `issue/75` branch (see https://github.com/mp911de/logstash-gelf/compare/issue/75). Can you give it a try to check whether it solves your issue?

**Comment by FireballDWF** [comment]
Problem Solved, Big Thank You!

**Comment by mp911de** [comment]
Great, thanks for your feedback. The fix will be shipped with 1.9.0
